### DISCUSSION TITLE
Unable to drag/resize/etc window

### DISCUSSION THREAD

**Discussion by bantern** [audacious-media-player/audacious#105] [Q&A] [answered]
Hi all, my issue is that I am **unable to click and drag or resize the window for Audacious when using any Winamp skin, in either GTK (legacy) or Qt mode**. The window resizing issue is not there when using Qt interface. Launching from cli with verbose flags set doesn't seem to throw any meaningful errors, either.

I am on KDE neon 5.27, Plasma 5.27.10, KDE Frameworks 5.113.0, Qt version 5.15.11.

This is not a dealbreaker, but it would be nice to be able to use the Winamp skins, for the nostalgia/aesthetic reasons. Any ideas?

**Comment by radioactiveman** [comment]
Does KDE already use Wayland by default? If yes, have you tried if it works with X11 / XWayland?

**Reply by bantern** [reply]
I do not use Wayland, only X11, as I've found Wayland to be too buggy with my setup and desired configuration - ergo, this issue is occurring with X11

**Reply by bantern** [reply]
If it helps at all, I do not get a titlebar, but I think this may be default behavior, but I think it's why it's not working? I can't be certain though.

**Reply by radioactiveman** [reply]
A screenshot may help. But I assume we can't help you here and would advice to report this issue to KDE.

**Reply by bantern** [reply]
![Screenshot_20240123_180659](https://github.com/audacious-media-player/audacious/assets/20668537/54101a47-3d37-45d4-b37b-974d4c1732a3)

This is what the player looks like with "Winamp Classic Interface". The top-left menu button does work as expected, but the window cannot be moved. The visualizer windows _can_ be moved, though the playlist editor (not pictured) cannot be moved, just like the main window. 

![Screenshot_20240123_181132](https://github.com/audacious-media-player/audacious/assets/20668537/d8e9bd0a-a2ee-4d51-b384-8c0285b5d6c6)

When using the Qt Interface, I can move all windows with no issue. As an aside, and I think this is a KDE issue, the top-left menu appears at the top of my main display no matter where the window is positioned. I think it's KDE related because this is not the only program where this kind of issue occurs.

I don't understand how the Winamp skin not being able to move would be a KDE issue; this is the only program I've ever had an issue with control of a window before, so if you could inform me a bit on why that would be, it would be great knowledge to have. Thanks for your time spent replying to this issue!

**Reply by radioactiveman** [reply]
> I don't understand how the Winamp skin not being able to move would be a KDE issue.

Because moving the window(s) works and has worked for years with multiple desktop environments (including KDE as I am aware of) and regardless of GTK or Qt, ~~X11 or Wayland~~.

@jlindgren90: Are you aware of such issue on KDE? Do you have a KDE VM available by chance to test this?

**Reply by jlindgren90** [reply]
Dragging windows in the Winamp Classic Interface works fine for me under KDE Plasma 5.27.4 (X11). I would expect it not to work under native Wayland, since as a matter of policy, xdg-shell windows are not allowed to position themselves directly.

**Reply by radioactiveman** [reply]
Indeed, it does not work on Wayland. I have tested it right now on GNOME.

@bantern: What is the output of `echo $XDG_SESSION_TYPE`? I bet you use Wayland. :smiley:
Please also try `GDK_BACKEND=x11 audacious --gtk` or `QT_QPA_PLATFORM=xcb audacious` to run Audacious with Xwayland.

**Reply by bantern** [reply]
I can't try in Xwayland yet, I'll check back in to do so, but here's my proof I'm using X11 lol
![Screenshot_20240125_181220](https://github.com/audacious-media-player/audacious/assets/20668537/c98dd296-74ba-4856-a4f2-5403ccbba4ea)

**Comment by bantern** [comment]
Okay, I had some more time, here is what happened when running both the `GDK_BACKEND=x11 audacious --gtk` and `QT_QPA_PLATFORM=xcb audacious` commands:

`GDK_BACKEND=x11 audacious --gtk` threw a warning about a missing theme engine, started Audacious in the Winamp interface (it was closed in the Qt interface immediately prior to running the cmd), and I was still unable to move the window at all.
![Screenshot_20240126_074907](https://github.com/audacious-media-player/audacious/assets/20668537/14910ea2-fdba-4b79-8d6b-e340aa9e5010)

`QT_QPA_PLATFORM=xcb audacious` threw many errors about missing song files (this happens every time I launch - unrelated I think), but started Audacious in the Qt interface. When switched to Winamp interface from settings, I still could not move the window.
![Screenshot_20240126_081522-1](https://github.com/audacious-media-player/audacious/assets/20668537/5a5653fe-0266-46e3-b95a-bc193b80d6fd)

EDIT: Sorry for being annoying with info here, but just wanted to let you know that I figured out why it was throwing errors about missing files; I had erroneous .cue and .m3u files in some directories, so that's fixed.

**Reply by radioactiveman** [reply, marked as answer]
The two commands can be used to start Audacious in a Wayland session with XWayland. Using them in a X11 session makes no sense.

I have downloaded the KDE neon 5.27 live image today, started it in a virtual machine and installed Audacious. Moving the Winamp interface windows was working correctly with Qt and GTK.

There must be something wrong with your setup. We can't help you here unfortunately. Good luck, let us know when you found a solution.

**Comment by eugenialoli** [comment]
I hope some fix is found for this, because unless you know about these two environment variables, you can't move the winamp-style windows around using the window manager of Gnome. I'm using Debian-testing.

**Reply by jlindgren90** [reply]
You can provide this feedback to your preferred Linux vendor (you pay for support right?) and ask them to work on a fix for the issue.

As mentioned already, Wayland breaks existing applications by design, and there is currently no way around the design limitation other than using XWayland.

**Reply by radioactiveman** [reply]
We could set the environment variable like Qmmp does now: https://sourceforge.net/p/qmmp-dev/code/11358/

Do you think this is a reasonable approach, John?

**Reply by jlindgren90** [reply]
I am good with that. We would need to do it across-the-board (for both Qt and GTK) and not just for the skinned UI, because one can switch at runtime.

**Reply by radioactiveman** [reply]
I prepared a pull request, please check https://github.com/audacious-media-player/audacious/pull/1395.

**Reply by BRTPOB** [reply]
I just found (again) Audacious, and decided to try it out and I'm running into the same issue, however I'm not using Wayland, I'm using X11 on the Ubuntu derivative Pop!_OS with Gnome. For some reason I'm unable to build the 4.4.1 package as it's complaining about Qt6 not being installed though it is, so I'm using the flatpak version of 4.4.

The old 4.1 version that's prepackaged in the repo has the same issue, however I assume that's old enough to not have any bearing on this issue.

**Reply by radioactiveman** [reply]
For Ubuntu and derivatives, there is a PPA for Audacious which we recommend to use, see https://audacious-media-player.org/download.

If you want to build your own version, better use Qt 5 since Pop!_OS seems to be still based on Ubuntu 22.04. You can find the specific package names in [install-dependencies.sh](https://github.com/audacious-media-player/audacious-plugins/blob/master/.github/actions/install-dependencies/install-dependencies.sh).

**Reply by BRTPOB** [reply]
Awesome, I missed the repo so I updated to use that, as I do want the ability to get the newer builds as they come out. That said, I'm running 4.4.1 and I'm still unable to resize the window using the Winamp interface, with the Refugee skin specifically.

**Reply by radioactiveman** [reply]
Then this seems to be unrelated to the initial topic. Please file a separate bug report with more details and a short video.

Is GTK, Qt or both affected?
Does moving the window work?
Which window do you want to resize?

I'm not aware that GNOME is having issues regarding resizing Audacious' Winamp interface. @mschwendt: Can you confirm this?

**Comment by mschwendt** [comment]
With Wayland and Audacious Qt 5 Winamp mode one could not move the windows, but one could either set $QT_QPA_PLATFORM=xcb before running Audacious as a workaround, or switch to Gtk based Winamp skins UI, or select Xorg when logging in. Audacious 4.4 included related changes.

I'm not aware of any issues with resizing the Winamp skins. Ctrl+D double-size I guess.

Meanwhile it's Fedora 41 with Xwayland here, though.
$ xisxwayland && echo "yes"
yes
